packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/index.tsx

```diff
@@ -2,33 +2,80 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React, { useState } from 'react';
-import { RouteComponentProps } from '@reach/router';
+import React, { useEffect, useState } from 'react';
+import { navigate, RouteComponentProps, useLocation } from '@reach/router';
+import { REACT_ENTRYPOINT } from '../../../constants';
+import {
+  AuthUiErrors,
+  composeAuthUiErrorTranslationId,
+} from '../../../lib/auth-errors/auth-errors';
+import { logViewEvent, usePageViewEvent } from '../../../lib/metrics';
 import { FtlMsg } from 'fxa-react/lib/utils';
-import { useFtlMsgResolver } from '../../../models/hooks';
-import { usePageViewEvent } from '../../../lib/metrics';
-// import { useAlertBar } from '../../models';
+import {
+  useAccount,
+  useAlertBar,
+  useFtlMsgResolver,
+} from '../../../models/hooks';
+import AppLayout from '../../../components/AppLayout';
+import Banner, { BannerType } from '../../../components/Banner';
+import CardHeader from '../../../components/CardHeader';
 import FormVerifyCode, {
   FormAttributes,
 } from '../../../components/FormVerifyCode';
-import { REACT_ENTRYPOINT } from '../../../constants';
 import { MailImage } from '../../../components/images';
-import CardHeader from '../../../components/CardHeader';
-
-// email will eventually be obtained from account context
-export type ConfirmSignupCodeProps = { email: string };
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+import { Newsletter } from '../../../components/ChooseNewsletters/newsletters';
 
 export const viewName = 'confirm-signup-code';
 
-const ConfirmSignupCode = ({
-  email,
-}: ConfirmSignupCodeProps & RouteComponentProps) => {
+export type RouterState = {
+  email: string;
+  newsletters?: Newsletter[];
+};
+
+interface BannerOptions {
```

Comment:
Maybe we can use the `BannerProps` type from the Banner component and omit the `children` option for less duplication?

Reply:
Ah yeah, it took some finagling but I've figured out a way to make this work 👍